### ISSUE TITLE
Plasma 6 support

### ISSUE THREAD

**Issue by dagguh** [Meriemi-git/ColoredGauge#2]
On Plasma 6, the widget is not listed anymore.

![image](https://github.com/Meriemi-git/org.meriemi.coloredgauge/assets/1194704/b87ab24f-679b-4870-9a95-660dc0395b32)

Found this [guide for migrating from Plasma 5 to Plasma 6](https://develop.kde.org/docs/plasma/widget/porting_kf6/).

**Comment by Meriemi-git** [comment]
Hi ! Thanks for the issue and guide, i have already porting the majority of the code and i do a lot of rework, codebase was ugly in fact. I added few interesting features but for now i'm really busy IRL so i will update the widget in couple of weeks. I hope you will enjoy the new features.

**Comment by dagguh** [comment]
Amazing, thank you! It's a great widget :slightly_smiling_face:

**Comment by Meriemi-git** [comment]
I have update the widget, you can find it on KDE store from now, you need to reinstall it to use the Plasma 6 compatible version.

**Comment by dagguh** [comment]
Uninstall + reinstall worked partially. I got the sensor display style on the list, I could configure it (a ton of new options, beautiful!).
But after filling everything in, and pressing OK, the widget disappears.

Then I tried to uninstall, reboot and install. Now the installation looks differently:
![image](https://github.com/Meriemi-git/org.meriemi.coloredgauge/assets/1194704/39874b50-b0b7-4cb6-9b24-48b19248869a).
First I tried the tar.g.z with the yesterdays' date in it, but nothing happens. Same with the other file. I'll try rebooting again.

**Comment by dagguh** [comment]
Second reboot, the style is not on the list. The installation dialog gives the same choice of tar.gz, but the install doesn't do anything. Should I attach some logs?

**Comment by dagguh** [comment]
After a qt5 update via pamac, it installed!
![image](https://github.com/Meriemi-git/org.meriemi.coloredgauge/assets/1194704/4e4d6391-c30c-40e2-9665-c28667f096cb)
At first the CPU one was very wide, so I fiddled with negative paddings. Then the GPU one was looking right from the get go. I've reset the CPU to 0 padding and all the other settings are the same, but they look different. Oh well, doesn't matter, they're beautiful anyway :)
Recreated the CPU one from scratch, now it's perfect :)
![image](https://github.com/Meriemi-git/org.meriemi.coloredgauge/assets/1194704/ece410b7-7be9-447b-8e11-d6d423f37f1b)

**Comment by dagguh** [comment]
PS. If I try to use the gauge for RPM, the max data range is 99, but fans reach 1400 RPM.

**Comment by Meriemi-git** [comment]
> After a qt5 update via pamac, it installed! ![image](https://private-user-images.githubusercontent.com/1194704/339271635-4e4d6391-c30c-40e2-9665-c28667f096cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzMDg3MzgsIm5iZiI6MTcxODMwODQzOCwicGF0aCI6Ii8xMTk0NzA0LzMzOTI3MTYzNS00ZTRkNjM5MS1jMzBjLTQwZTItOTY2NS1jMjg2NjdmMDk2Y2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTNUMTk1MzU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODU3MDEzZjQ0MmY5YTI5Nzk2MWY2YTVhNjlhMmEzYjhiYjhjZTUzNzUwNDBlNzU2Y2RmN2YyZDljZTI3NTkzNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aXes6rtkZTm3RP280tzKlbw6-sGZ9emPZwgxAYj3EK8) At first the CPU one was very wide, so I fiddled with negative paddings. Then the GPU one was looking right from the get go. I've reset the CPU to 0 padding and all the other settings are the same, but they look different. Oh well, doesn't matter, they're beautiful anyway :) Recreated the CPU one from scratch, now it's perfect :) ![image](https://private-user-images.githubusercontent.com/1194704/339272927-ece410b7-7be9-447b-8e11-d6d423f37f1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgzMDg3MzgsIm5iZiI6MTcxODMwODQzOCwicGF0aCI6Ii8xMTk0NzA0LzMzOTI3MjkyNy1lY2U0MTBiNy03YmU5LTQ0N2ItOGUxMS1kNmQ0MjNmMzdmMWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTNUMTk1MzU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTRkMDUyMzk4OGRlNTZmMTUzNDdlNWQ0MGEwYTI3OGU2N2RjMzQ3Njk2MDA5ZjIzMGFjYjE3MWRiZTYzNzdhMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2QyswrX2Qe8B1h8XZ293vEF9JhYI9_QSaVEw5XNXetc)

On initialization the width of the widget is determined by the length of the title. But you can force the width with the minimum width setting.
If you adjust padding and width it can be usefull to refresh the plasmashell layout with this CMD :
` kquitapp5 plasmashell > /dev/null 2>&1 ; kstart5 plasmashell > /dev/null 2>&1`

**Comment by Meriemi-git** [comment]
> PS. If I try to use the gauge for RPM, the max data range is 99, but fans reach 1400 RPM.

Have you try to set the range by hand inside the data tab ?

**Comment by dagguh** [comment]
> On initialization the width of the widget is determined by the length of the title.

True. After yet another reboot, suddenly 6 extra widgets appeared. Some were from my failed attempts after the update. Some were back from Plasma 5 times :)
Widths were determined by titles, now all is in good order.

**Comment by dagguh** [comment]
> Have you try to set the range by hand inside the data tab ?

Yes, but the max range inside the data tab is 99.

**Comment by Meriemi-git** [comment]
I will check
